### PR TITLE
setup.cfg: added flake8 settings

### PR DESCRIPTION
**Pull request by back-to**

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -11,3 +11,29 @@ versionfile_source = src/streamlink/_version.py
...
+  .git/,
+  build/,
+  dist/,
+  docs/,
+  env/,
+  src/streamlink/packages/flashmedia/,
+  venv/,
+  versioneer.py,
+  win32/,
+max-line-length = 128
```
Comment (beardypig) [inline setup.cfg]: 👍

**Comment by gravyboat** [comment]
I'm good with this. Let's see if anyone else weighs in, mostly @bastimeyer.

**Comment by bastimeyer** [comment]
Is the intention of this PR that editors/IDEs can pick up a common linting config? Because that's just the linting config, the actual linter is not being used in the CI.

IIRC, a linter was never added after the fork, because the argument back then was that it was not important at that time. Now, if we add a linting config, then we should also enable it in the CI, so that every PR can be checked.

However, if we add it now, then a lot of work needs to be done first (similar to #414). I'm not sure about the Python-world, but do these linting tools here support automatic code fixing? (I usually don't trust auto-fixing stuff though)

The longer we wait, the more work needs to be done in the future. We could allow linting failures for now though, but I'm not sure how useful it would be, especially when checking new PRs.

```
$ flake8 --statistics --count --quiet --quiet
3     E117 over-indented
3     E121 continuation line under-indented for hanging indent
1     E122 continuation line missing indentation or outdented
13    E126 continuation line over-indented for hanging indent
3     E127 continuation line over-indented for visual indent
26    E128 continuation line under-indented for visual indent
2     E203 whitespace before ':'
9     E226 missing whitespace around arithmetic operator
18    E241 multiple spaces after ':'
2     E265 block comment should start with '# '
4     E302 expected 2 blank lines, found 1
23    E303 too many blank lines (3)
5     E305 expected 2 blank lines after class or function definition, found 1
3     E402 module level import not at top of file
106   E501 line too long (245 > 128 characters)
1     E711 comparison to None should be 'if cond is not None:'
3     E712 comparison to True should be 'if cond is not True:' or 'if not cond:'
2     E722 do not use bare 'except'
10    E741 ambiguous variable name 'l'
71    F401 '.api.streams' imported but unused
4     F403 'from unittest.mock import *' used; unable to detect undefined names
28    F405 'Stream' may be undefined, or defined from star imports: streamlink.stream
2     F601 dictionary key 'rtmp' repeated with different values
1     F811 redefinition of unused 'test_expired_at' from line 59
5     F821 undefined name 'self'
22    F841 local variable 'e' is assigned to but never used
10    W291 trailing whitespace
1     W292 no newline at end of file
2     W293 blank line contains whitespace
6     W391 blank line at end of file
26    W504 line break after binary operator
415
```

**Comment by back-to** [comment]
> Is the intention of this PR that editors/IDEs can pick up a common linting config?

yes

> then we should also enable it in the CI, so that every PR can be checked.

that could be done in the future, but this PR is just for the editors/IDEs

**Comment by gravyboat** [comment]
Are we good to merge this? Is something still holding it up?

**Comment by beardypig** [comment]
I'm OK with it.

**Comment by gravyboat** [comment]
Merging then. Thanks @back-to!

**Pull request merged**